### ISSUE TITLE
Missing error for invalid 'scoped' modifier on return type (including no errors in C# 10)

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#62645]
**Version Used**: 
```
commit a428264e50d662bc0968cbb2c28b3a55df092f22
Author: Charles Stoner <10732005+cston@users.noreply.github.com>
Date:   Wed Jun 22 18:43:39 2022 -0700

    Ref fields escape analysis (#61195)
```

**Steps to Reproduce**:

Compile the following code with C# 10 and C# 11
```cs
ref struct RefStruct { }

class C { public ref scoped RefStruct M() => throw null!; }
delegate ref scoped RefStruct D3();
```

**Expected Behavior**:
`error CS0106: The modifier 'scoped' is not valid for this item` is reported for the `scoped` modifier

**Actual Behavior**:
No errors is reported; the code is compiled successfully with both C# 10 and C# 11

**Notes**:
- In C# 10 the code shouldn't be valid at all as scoped modifiers are only supported as a part of ref fields which are limited to C# 11
- In C# 11 there should be an error about invalid modifier usage, as scoped modifiers do not affect return types and in this case are lost during compilation since LiftimeAnnotationAttribute is not emitted for return value

**Comment by cston** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/62088.
